src/node/anvil.rs

```diff
@@ -0,0 +1,85 @@
+use zksync_basic_types::{Address, U256, U64};
+use zksync_web3_decl::error::Web3Error;
+
+use crate::{
+    fork::ForkSource,
+    namespaces::{AnvilNamespaceT, ResetRequest, RpcResult},
+    node::InMemoryNode,
+    utils::{into_jsrpc_error, into_jsrpc_error_message, IntoBoxedFuture},
+};
+
+impl<S: ForkSource + std::fmt::Debug + Clone + Send + Sync + 'static> AnvilNamespaceT
+    for InMemoryNode<S>
+{
+    fn set_balance(&self, address: Address, balance: U256) -> RpcResult<bool> {
+        self.set_balance(address, balance)
+            .map_err(|err| {
+                tracing::error!("failed setting balance : {:?}", err);
+                into_jsrpc_error(Web3Error::InternalError(err))
+            })
+            .into_boxed_future()
+    }
+
+    fn set_nonce(&self, address: Address, balance: U256) -> RpcResult<bool> {
+        self.set_nonce(address, balance)
+            .map_err(|err| {
+                tracing::error!("failed setting nonce: {:?}", err);
+                into_jsrpc_error(Web3Error::InternalError(err))
+            })
+            .into_boxed_future()
+    }
+
+    fn hardhat_mine(&self, num_blocks: Option<U64>, interval: Option<U64>) -> RpcResult<bool> {
```

Comment:
probably `anvil_mine`?

Reply:
Well, it's possible, but it doesn't really help, because if we rename it, we'll have `anvil_mine` implementation calling `hardhat_mine`...

Reply:
> we'll have anvil_mine implementation calling hardhat_mine...

That should be the correct approach, we should call `hardhat_mine` under the hood and not set it in the public interface like this:
```
    #[rpc(name = "anvil_mine")]
    fn hardhat_mine(&self, num_blocks: Option<U64>, interval: Option<U64>) -> RpcResult<bool>;

```